Core/Tests/Sources/Model/EnhancedLocationUpdateTests.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+import CoreLocation
+import XCTest
+@testable import Core
+
+class EnhancedLocationUpdateTests: XCTestCase {
+
+    func testEnhancedLocationUpdateMessageDecodesNullBatteryLevel() throws {
+        let location = CLLocation(latitude: 0.0, longitude: 0.0)
+        let expected = try! EnhancedLocationUpdateMessage(locationUpdate: EnhancedLocationUpdate(location: location), batteryLevel: nil)
+        
+        let jsonString = try! expected.toJSONString()
+        let actual: EnhancedLocationUpdateMessage = try! EnhancedLocationUpdateMessage.fromJSONString(jsonString)
+        
+        XCTAssertEqual(expected.batteryLevel, actual.batteryLevel)
+        XCTAssertEqual(expected.type, actual.type)
+        
+// There are more things to XCTAssertEqual but you'll need to implement Equatable or start using Structs instead of classes.
+//        XCTAssertEqual(expected.location, actual.location)
+//        XCTAssertEqual(expected.intermediateLocations, expected.intermediateLocations)
```

Comment:
I'm wondering if we should have those other assertions here as the purpose of this test is to check the `batteryLevel` field :thinking: But I'll leave it up to you :wink:

Reply:
I've deleted this test because by removing the battery level completely, there is nothing to test. Let me know if you think there are some tests I should write, thanks :)